source/geometry_model/interface.cc

```diff
@@ -36,6 +36,19 @@ namespace aspect
     {}
 
 
+    template <int dim>
+    void
+    Interface<dim>::update_surface ()
+    {}
+
+
+    template <int dim>
+    double
+    Interface<dim>::depth_including_mesh_deformation(const Point<dim> &position) const
+    {
+      return depth(position);
+    }
```

Comment:
You want to throw an exception here. Models that don't implement this function shouldn't return wrong values, but just abort.

Reply:
I've added an assertthrow.